py/torch_tensorrt/dynamo/runtime/_PythonTorchTensorRTModule.py

```diff
@@ -111,6 +111,41 @@ def __init__(
         if self.serialized_engine is not None and not self.settings.lazy_engine_init:
             self.setup_engine()
 
+    def get_streamable_weights_size(self) -> Any:
+        return self.engine.streamable_weights_size
+
+    def get_automatic_weight_streaming_budget(self) -> Any:
+        return self.engine.get_weight_streaming_automatic_budget()
+
+    def get_device_memory_budget(self) -> Any:
+        return self.engine.weight_streaming_budget_v2
+
+    def set_device_memory_budget(self, budget_bytes: int) -> int:
+        # Recreating the context because weight streaming budget cannot be modified while there are active context.
+        if self.context is not None:
+            del self.context
+        budget_bytes = self._set_device_memory_budget(budget_bytes)
+        self.context = self.engine.create_execution_context()
+        return budget_bytes
+
+    def _set_device_memory_budget(self, budget_bytes: int) -> int:
+        # Disable weight streaming for invalid budget size
+        if budget_bytes < 0:
+            budget_bytes = self.get_streamable_weights_size()
+        self.engine.weight_streaming_budget_v2 = budget_bytes
+        if self.engine.weight_streaming_budget_v2 != budget_bytes:
```

Comment:
when would this condition fail ?

Reply:
if invalid value(negative, >streamable size) is set, this condition fails.